src/libraries/Common/src/System/Net/Security/Unix/SafeFreeCredentials.cs

```diff
@@ -27,27 +27,21 @@ protected SafeFreeCredentials(IntPtr handle, bool ownsHandle) : base(handle, own
     //
     // This is a class holding a Credential handle reference, used for static handles cache
     //
-#if DEBUG
-    internal sealed class SafeCredentialReference : DebugCriticalHandleMinusOneIsInvalid
-    {
-#else
-    internal sealed class SafeCredentialReference : CriticalHandleMinusOneIsInvalid
+    internal sealed class SafeCredentialReference : CriticalFinalizerObject, IDisposable
     {
-#endif
-
         //
         // Static cache will return the target handle if found the reference in the table.
         //
-        internal SafeFreeCredentials Target;
+        internal SafeFreeCredentials Target { get; private set; }
```

Comment:
Why did you make this a property here but not in the other copy?

Reply:
I merged the SafeFreeCredentials class so it is consistent now